Mathlib/Analysis/Calculus/TaylorIntegral.lean

```diff
@@ -0,0 +1,164 @@
+      iteratedFDeriv_zero_apply, one_smul, pow_zero, reduceAdd, iteratedFDeriv_one_apply]
+    rw [← sub_eq_iff_eq_add', Eq.comm, intervalIntegral.integral_congr h_eq]
+    have hf' : ∀ (t : ℝ) (ht : t ∈ Set.uIcc 0 1), DifferentiableAt ℝ (fun s ↦ f (x + s • y)) t :=
+      fun t ht ↦ ((hf t ht).differentiableAt (by simp)).comp t (by fun_prop)
+    have hint : IntervalIntegrable (deriv (fun s ↦ f (x + s • y))) MeasureTheory.volume 0 1 := by
+      have h₁ : ContinuousOn (fderiv ℝ f) ((fun t ↦ x + t • y) '' Set.uIcc (0 : ℝ) 1) := by
+        intro z ⟨t, ht, hz⟩
+        rw [← hz]
+        exact (((hf t ht).fderiv_right (le_refl _)).continuousAt (n := 0)).continuousWithinAt
+      have h₂ : ContinuousOn (fun x_1 ↦ fderiv ℝ f (x + x_1 • y)) (Set.uIcc (0 : ℝ) 1) := by
+        apply h₁.comp (t := (fun t ↦ x + t • y) '' (Set.uIcc (0 : ℝ) 1)) (by fun_prop)
+        intro t ht
+        use t
+      apply (ContinuousOn.congr _ h_eq.symm).intervalIntegrable
+      fun_prop
+    simpa using intervalIntegral.integral_deriv_eq_sub hf' hint
+  | succ n ih =>
+    -- We use the inductive hypothesis to cancel all lower order terms
+    specialize ih (fun t ht ↦ (hf t ht).of_le (by simp))
+    rw [Finset.sum_range_succ, add_assoc]
+    convert ih using 2
+    -- We define the functions u and v that we will integrate by parts
+    set u := fun (k : ℕ) (t : ℝ) ↦ (k ! : ℝ)⁻¹ * (1 - t) ^ k
+    have hu : ∀ (t : ℝ), HasDerivAt (u (n + 1)) (-u n t) t := by
+      intro t
+      unfold u
+      have : (-((n ! : ℝ)⁻¹ * (1 - t) ^ n)) =
+          ((n + 1) ! : ℝ)⁻¹ * ((n + 1) * (1 - t) ^ n * (-1)) := by
+        field_simp
+        congr 1
+        rw [Nat.factorial_succ]
+        grind
+      rw [this]
+      convert (((hasDerivAt_id t).const_sub 1).pow _).const_mul _
+      norm_cast
+    have hu' : Continuous (u n) := by fun_prop
+    set v := fun (k : ℕ) (t : ℝ) ↦ iteratedFDeriv ℝ k f (x + t • y) (fun _ ↦ y)
+    have hv : ∀ (t : ℝ) (ht : t ∈ Set.uIcc 0 1), HasDerivAt (v (n + 1)) (v (n + 1 + 1) t) t := by
+      intro t ht
+      unfold v
+      rw [← (hf t ht).deriv_fderiv_add_smul]
+      have h_diff : DifferentiableAt ℝ (iteratedFDeriv ℝ (n + 1) f) (x + t • y) := by
+        apply (hf t ht).differentiableAt_iteratedFDeriv
+        norm_cast
+        grind
+      refine DifferentiableAt.hasDerivAt ?_
+      fun_prop
+    have hv' : ContinuousOn (v (n + 1 + 1)) (Set.uIcc 0 1) := by
+      intro t ht
+      have h_cont : ContinuousAt (iteratedFDeriv ℝ (n + 1 + 1) f) (x + t • y) :=
+        ((hf t ht).iteratedFDeriv_right (i := n + 1 + 1) (m := 0) (by simp)).continuousAt
+      exact (h_cont.comp (x := t) (by fun_prop)).continuousWithinAt.eval_const _
+    -- Now we apply integration by parts and simplify
+    simpa [← eq_neg_add_iff_add_eq, ← intervalIntegral.integral_smul, smul_smul, u, v] using
+      intervalIntegral.integral_smul_deriv_eq_deriv_smul (fun t _ ↦ hu t) hv
+      (hu'.neg.intervalIntegrable _ _) hv'.intervalIntegrable
+
```

Comment:
I'd be happy to deduce from the above the version assuming `ContDiffOn f (closedBall x (||y||))`, but I don't think this will be straightforward (because under this assumption the function is not `ContDiffAt f (x+y)`). Do you see an easy way forward? (Maybe taking a limit of the previous lemma applied to (1-\epsilon)y)?